libparsec/crates/client/src/workspace/store/file_updater.rs

```diff
@@ -134,9 +134,11 @@ pub(super) async fn for_update_file(
 
     let manifest = match manifest {
         ArcLocalChildManifest::File(manifest) => manifest,
-        ArcLocalChildManifest::Folder(_) => {
+        ArcLocalChildManifest::Folder(manifest) => {
             release_guard_on_error!(update_guard);
-            return Err(ForUpdateFileError::EntryNotAFile);
+            return Err(ForUpdateFileError::EntryNotAFile {
```

Comment:
**nitpick**: If find that we loose information about the type of the related entry id

I think it could be nice to have an additional field like `got: ManifestType`.

That would improve the error message:

`Entry id <id> is not a file but a <type>`

Reply:
Currently there is only two types of entry: file and folder, so this doesn't bring much information ^^

I think the important thing here is to keep in mind the primary goal of this error (i.e. "the operation failed because the entry wasn't a file").
This is for this reason I avoided naming this error `EntryISAFolder`, which would convey the same information (since there is only file or folder), but is less clear given here we actually care about file, not folder.
It is always tempting to add more information to an error (especially if there are easily available in the function creating the error !), but I think it makes things more complicated for no reason.

However the current situation is interesting:
- winfsp needs a way to open a file or a folder
- but we can only open files in libparsec API, hence "opening" a folder only consist of retrieving its entry ID
- opening the file or getting the entry ID if it's a folder should be done in an atomic way, otherwise an ugly loop is needed to handle the case the entry is first a folder when we try to open it as a file, then becomes a file consider it as folder and try to get it entry ID :/

So in this situation, going against the rule I just stated is very appealing since it makes thing a bit less coherent in theory, but a lot better in practice !
This is the kind of trade-of that can only be judged on the spot, hence why I don't think we should add more information than needed.